File: codegen/results/v002/qwen3.8_27b_nothink/20260827_161044/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputPath = path.join('input', 'text.txt');
const text = fs.readFileSync(inputPath, 'utf-8');

// Convert to lowercase, keep only letters (strip punctuation)
const words = text
  .toLowerCase()
  .replace(/[^a-z\s]/g, ' ')  // replace non-letter, non-whitespace with space
  .split(/\s+/)
  .filter(w => w.length > 0);

const freq = new Map<string, number>();
for (const word of words) {
  freq.set(word, (freq.get(word) || 0) + 1);
}

// Convert to array and sort
const entries = Array.from(freq.entries());
entries.sort((a, b) => {
  if (b[1] !== a[1]) {
    return b[1] - a[1];
  }
  // Alphabetical ascending
  return a[0].localeCompare(b[0]);
});

for (const [word, count] of entries) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v002/qwen3.8_27b_nothink/20260827_161044/python/002_word_frequency/solution.py
import re

def main():
    with open('input/text.txt', 'r') as f:
        text = f.read()
    
    # Convert to lowercase
    text = text.lower()
    
    # Keep only letters (strip punctuation)
    # We want to extract words consisting only of letters
    words = re.findall(r'[a-z]+', text)
    
    # Count frequencies
    freq = {}
    for word in words:
        freq[word] = freq.get(word, 0) + 1
    
    # Sort by count descending, then alphabetically ascending for ties
    sorted_words = sorted(freq.items(), key=lambda x: (-x[1], x[0]))
    
    # Output
    for word, count in sorted_words:
        print(f"{word}: {count}")

if __name__ == '__main__':
    main()